libs/util/httpRequest.js

```diff
@@ -121,15 +121,19 @@ function httpRequest(options, attempt) {
             options.retry.interval *
             Math.pow(2, currentAttempt);
 
+        controller = new AbortController();
+        currentAttempt += 1;
         return delayPromise(function () {
-            return httpRequest(options, currentAttempt + 1);
+            return io(options, controller).catch(retry);
         }, delay);
     });
 
     return {
         then: promise.then.bind(promise),
         catch: promise.catch.bind(promise),
-        abort: controller.abort.bind(controller),
+        abort: function () {
+            return controller.abort();
+        },
```

Comment:
@redonkulus it was necessary to create a proper function since it might be that the initial `controller` is already stale. So, here, whenever users calls `abort` we are always using a fresh reference.

Reply:
lets add this comment in the code for future reference